articles/server-platforms/rails/02-custom-login.md

```diff
@@ -26,15 +26,15 @@ ${snippet(meta.snippets.dependencies)}
 
 ```ruby
 <div id="login-container">
-  <%= form_tag do %>
-      <%= text_field_tag :user, "", class:'form-control' %>
+  ${ '<%= form_tag do %>' }
+      ${ '<%= text_field_tag :user, "", class:"form-control" %>' }
       <br>
-      <%= password_field_tag :password, "", class:'form-control'%></p>
+      ${ '<%= password_field_tag :password, "", class:"form-control"%>' }</p>
       <br>
-      <%= submit_tag("Login", class: "btn btn-success btn-lg") %>
-      <%= submit_tag("Sign up", name: "signup", class:"btn btn-success btn-lg") %>
+      ${ '<%= submit_tag("Login", class: "btn btn-success btn-lg") %>' }
+      ${ '<%= submit_tag("Sign up", name: "signup", class:"btn btn-success btn-lg") %>' }
   <% end %>
```

Comment:
@fleko 
This line throws the application error, it needs to be escaped as well:
 `<% end %>` --> `${ '<% end %>' }`